jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1577,15 +1577,18 @@ YUI.add('juju-view-utils', function(Y) {
       return;
     }
 
-    jem.newModel(
-      users.jem.user,
+    controllerAPI.createModel(
       model,
-      credential,
+      // Check that this is a user tag, if not turn it into one.
+      cloud.indexOf('user-') === 0 ? user : `user-${user}`,
       {
-        cloud: cloud,
+        // Check that this is a credential tag, if not turn it into one.
+        credentialTag: credential.indexOf('cloudcred-') === 0 ?
```

Comment:
Same, let's say we receive a credential tag, and let's ensure we always pass tags, and let's remove this logic please.